core/rule_manager.py

```diff
@@ -80,16 +80,17 @@ def list_rules(self, user_id: str) -> list:
         visited_rule_ids = []
         for rule in rules:
             if rule[0] not in visited_rule_ids:
-                res.append({
-                    'id': rule[0],
-                    'key': rule[1],
-                    'name': rule[2],
-                    'pattern': rule[3],
-                    'constraints': rule[4],
-                    'rewrite': rule[5],
-                    'actions': rule[6],
-                    'enabled_apps': rule_applications[rule[0]]
-                })
+                if app_id == None or (len(rule_applications[rule[0]]) and int(app_id) in [rule_applications[rule[0]][i]["app_id"] for i in range(len(rule_applications[rule[0]]))]):
```

Comment:
This line is too complicated for ohters to read and understand. Can you break the filtering logic into smaller steps? I understand this code is consice and smart, however, it is more important for future developers to read and understand the code.